hibernate-core/src/main/java/org/hibernate/engine/jdbc/batch/internal/BatchBuilderImpl.java

```diff
@@ -61,9 +61,15 @@ public void setJdbcBatchSize(int size) {
 
 	@Override
 	public Batch buildBatch(BatchKey key, JdbcCoordinator jdbcCoordinator) {
-		LOG.tracef( "Building batch [size=%s]", size );
-		return size > 1
-				? new BatchingBatch( key, jdbcCoordinator, size )
+		Integer batchSize = jdbcCoordinator.getJdbcSessionOwner().getBatchSize();
+		if(batchSize != null) {
+			LOG.tracef( "Building batch [Session batchSize=%s]", batchSize );
+		} else {
```

Comment:
Please update your IDE code style.  Again the Hibernate style is to put the `}' on its own line.